Multi/release v1.0.1

alinmigea opened the pull request:
@plentymarkets/team-multichannel-koboldmaki

Commit:
ADD adjustments for billiger.de, new set of properties fields were added in the export
ADD use of helper classes from ElasticSearch plugin
ADD the dependency for the ElasticExport plugin

Commit:
FIX the cache for the shipping costs
REMOVE calculating time logs for each row

Commit:
FIX for the property helper for the selection and text properties

mariusbippig commented on meta/documents/changelog_de.md:
```diff
@@ -1,5 +1,13 @@
+- Das Plugin Elastic Export ist nun Voraussetzung zur Nutzung des Pluginformats BilligerDE.
+
+### Hinzugefügt
+- Zusätzliche Felder in der Export-csv-Datei hinzugefügt, um die zusätzlichen Eigenschaften für billiger.de zugänglich zu machen.
```
Die Felder der CSV-Datei wurden erweitert, um neue Merkmale für billiger.de zu unterstützen.

mariusbippig commented on meta/documents/changelog_en.md:
```diff
@@ -1,5 +1,13 @@
+- The plugin Elastic Export is now required to use the plugin format BilligerDE.
+
+### Added
+- Added optional fields in the export csv file to provide the extra properties needed for billiger.de.
```
The fields of the CSV file were extended to support new properties for billiger.de.

mariusbippig commented on src/Generator/BilligerDE.php:
```diff
@@ -213,59 +264,61 @@ private function head():array
-                ]);
-            }
+            // Price for base price calculation
+            $variationRetailPrice['variationRetailPrice.price'] = $priceList['price'];
```
You don't need to set that in a new variable. Just use $priceList

mariusbippig commented on src/Generator/BilligerDE.php:
```diff
@@ -213,59 +264,61 @@ private function head():array
+                'desc'          => $this->elasticExportHelper->getMutatedDescription($variation, $settings),
+                'shop_cat'      => $this->elasticExportHelper->getCategory((int)$variation['data']['defaultCategories'][0]['id'], $settings->get('lang'), $settings->get('plentyId')),
+                'price'         => $priceList['price'],
+                'ppu'           => $this->elasticExportHelper->getBasePrice($variation, $variationRetailPrice, $settings->get('lang')),
```
There is now a ElasticExportPriceHelper, use that instead of base helper pls.

mariusbippig commented on src/Helper/PropertyHelper.php:
```diff
@@ -74,7 +81,7 @@ public function getProperty($variation, KeyValue $settings, string $property):st
      * @return array
      */
-    private function getItemPropertyList($variation, $lang = 'de'):array
+    private function getItemPropertyList($variation, $lang):array
```
Why have you converted the optional parameter to a required one?

Commit:
UPDATE the base price, now base price is calculated via ElasticExportPriceHelper
FIX the property helper propertyList method
FIX the keyMutator, now the keyMutator overwrites the skus array with default null, if none was found

hschwab committed:
Update changelog_de.md

Commit:
FIX the release date
